src/dosext/mfs/lfn.c

```diff
@@ -1328,6 +1328,11 @@ int mfs_lfn(void)
 {
 	int carry, ret;
 
+	if (!mach_fs_enabled) {
+		CARRY;
+		return 0;
+	}
+
```

Comment:
I don't think it is easy enough to make
it generic. Currently it looks into mfs's
drive[] array to find the linux path and
then uses posix FS api for manipulating
files and dirs. I think this is quite bad
design, but will you rewrite the half of
lfn.c to make it to use the redirector API
as a backend, with the ability to write
also an int13 backend for vfat?
If you do than, then of course it will have
the new home. But currently it hacks
around mfs, so you can't easily move
it away.
So now as I understand this, I suppose
there is no other way than the "big switch
off" as you proposed initially. It simply
doesn't work without mfs, sigh.